heimdall-middleware-spec/src/test/java/br/com/conductor/heimdall/middleware/util/helpermock/call/RequestMockTest.java

```diff
@@ -0,0 +1,79 @@
+package br.com.conductor.heimdall.middleware.util.helpermock.call;
+
+import br.com.conductor.heimdall.middleware.spec.Header;
+import br.com.conductor.heimdall.middleware.spec.Query;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.Timeout;
+import static org.junit.Assert.*;
+
+import java.util.HashMap;
+
+public class RequestMockTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+  @Rule public final Timeout globalTimeout = new Timeout(10000);
+
+  @Test
+  public void testHeaderOutputNotNull() {
+    final RequestMock objectUnderTest = new RequestMock();
+
+    final Header retval = objectUnderTest.header();
+
+    assertNotNull(retval);
+    assertEquals(new HashMap<String, String>(), retval.getAll());
+  }
+
+  @Test
+  public void testQueryOutputNotNull() {
+    final RequestMock objectUnderTest = new RequestMock();
+
+    final Query retval = objectUnderTest.query();
+
+    assertNotNull(retval);
+    assertEquals(new HashMap<String, String>(), retval.getAll());
+  }
+
+  @Test
+  public void testSetAppNameInput() {
+    final RequestMock objectUnderTest = new RequestMock();
+
+    objectUnderTest.setAppName("testName");
+
+    assertEquals("testName", objectUnderTest.getAppName());
+  }
+
+  @Test
+  public void testSetBodyInput() {
+    final RequestMock objectUnderTest = new RequestMock();
+
+    objectUnderTest.setBody("testBody");
+
+    assertEquals("testBody", objectUnderTest.getBody());
+  }
+
+  @Test
+  public void testSetSendResponse() {
+    final RequestMock objectUnderTest = new RequestMock();
+    objectUnderTest.setSendResponse(false);
+
+    // Method returns void, testing that no exception is thrown
```

Comment:
We don't usually test methods which don't return a value.

Reply:
This should be tested, but you have to check for an observable side effect.